packages/vue-components/src/utils/urls.js

```diff
@@ -1,10 +1,9 @@
 export default function normalizeUrl(url) {
-  if (url.endsWith('.html')) {
-    return url.toLowerCase();
-  } else if (url.endsWith('/')) {
-    return `${url}index.html`.toLowerCase();
-  } else if (!url.endsWith('/')) {
-    return `${url}.html`.toLowerCase();
+  const urlLower = url.toLowerCase();
+  if (urlLower.endsWith('.html')) {
+    return urlLower;
+  } else if (urlLower.endsWith('/')) {
+    return `${urlLower}index.html`;
   }
-  return url.toLowerCase();
+  return `${urlLower}/index.html`;
```

Comment:
Can I check why extensionless urls need to be appended with `/index.html`? I believe this breaks the `isExact` method because on our netlify deployment, the `.html` is omitted from the url on some of the pages.

For example this page, https://deploy-preview-1700--markbind-master.netlify.app/userguide/addingpages, the site nav corresponding url is `/userGuide/addingPages.html` and after normalizing, the href will remain as it is but the url will be appended to `/userguide/addingpages/index.html`.

![Capture](https://user-images.githubusercontent.com/47470981/147626747-62633dd0-39b5-46d8-9404-90fdbffe67ce.PNG)

Reply:
Thanks for highlighting this, completely missed it.

Was thinking solely of:
`https://markbind.org/userGuide` -> `https://markbind.org/userGuide/index.html`

But its also certainly possible for it to be interpreted as `https://markbind.org/userGuide.html`. (this is not the case for `markbind s`'s local server however (404))

I think I'll discard this commit / change for now, and just stick with the first one in this PR since the logic for this case is quite a bit more complex than initially thought.

We may need to consider such `href`s (without extension and without ending `/`) twice, interpreted once as `userGuide.html` and another time as `userGuide/index.html` in some way. The order of consideration of these links should also be thought through as well.

If there's no solution, the status quo might be better too, since `https://markbind.org/userGuide` -> `https://markbind.org/userGuide.html` is more *probably* likely than `https://markbind.org/userGuide` -> `https://markbind.org/userGuide/index.html`.

Reply:
Yup we can further explore on this. 

> If there's no solution, the status quo might be better too, since https://markbind.org/userGuide -> https://markbind.org/userGuide.html is more probably likely than https://markbind.org/userGuide -> https://markbind.org/userGuide/index.html.

I'm fine with this as well